docs/discussion/idioms/unit-expression-api.md

```diff
@@ -0,0 +1,107 @@
+# Unit Expression APIs
+
+Many APIs in the library accept a _Unit expression_, `u`.  Common examples include (for some
+Quantity `q` and [arithmetic](https://en.cppreference.com/w/cpp/types/is_arithmetic) type `T`):
+
+- `q.as(u)`
+- `round_as(u, q)`
+- `inverse_as(u, q)`
+- "Explicit-Rep" versions of the above (such as `q.as<T>(u)`)
+- "`in`" versions of the above, which exit the library (`q.in(u)`, `round_in(u, q)`, etc.)
```

Comment:
This was the most challenging piece of feedback to incorporate.  I really did want this page to be about the slot, not the one particular way of filling it.  Ultimately, I decided it made better sense as a "unit slots" page.

This led to downstream changes everywhere I discussed unit expressions.  I kept all these changes in a single commit, af12c478d0066c438268b1001ab12432572227da, for easier reviewability.  (The preceding commit does a raw move of the file to its new name, so that git could tell that it was a rename and not an add/delete pair.  This makes the diffs correct on the linked commit.)

Reply:
This now flows in a way that makes much more sense to my mind. Thank you!